### PR TITLE
chore(vercel): add react-router preview build and fix dir-index navigation

### PR DESCRIPTION
**Pull request by ShaneK**
Issue number: internal

---------

## What is the current behavior?

Currently, reviewing react router based PRs isn't possible using the vercel preview link

## What is the new behavior?

This PR adds support for the preview link to support react router 6 and tries to fix some weird edge cases where navigating to a route would sometimes not show the tests in that route

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

The new React Router preview app shares the React package gate (`REACT_PKG_OK`) because `@ionic/react-router` is built inside `build_react_pkgs()`. Test app builds run in parallel under that gate, then `wait` reports per-app failures.

Preview URLs (once Vercel builds this branch):

- React Router test app: https://ionic-framework-git-chore-vercel-add-rr-ionic1.vercel.app/react-router/
- Dir-index nav fix demo: https://ionic-framework-git-chore-vercel-add-rr-ionic1.vercel.app/src/components/progress-bar/test (no trailing slash; clicking `basic/`, `buffer/`, `determinate/` should now land on the test page instead of the root landing)

### CODE REVIEW

File: core/scripts/vercel-build.sh [hunk 1]
```diff
@@ -254,8 +262,29 @@ build_vue_test() {
...
+    echo "[react-router] No test app found, skipping."
+    return 0
+  }
+  echo "[react-router] Building ${APP}..."
+
+  cd "${REPO_ROOT}/packages/react-router/test"
+  ./build.sh "${APP}"
+  cd "build/${APP}"
+  # The react-router test apps require --legacy-peer-deps (mixed react versions
+  # across @ionic/react peer ranges).
```
Comment (thetaPC) [inline core/scripts/vercel-build.sh:277]: Assuming there's a ticket to remove the `--legacy-peer-deps`, let's add the TODO here.

**Review by thetaPC — APPROVED** [review]
LGTM, thank you!

File: core/scripts/vercel-build.sh (hunk 1 above)
Comment (ShaneK) [reply core/scripts/vercel-build.sh:277]: There's not, and I don't really want to. Any time we upgrade react router we'd have to add it back and there's no harm in having it at this point.
